src/audio/asrc/asrc.h

```diff
@@ -17,10 +17,42 @@
  * NOTE: This ABI uses bit fields and is non portable.
  */
 
-#ifndef __SOF_IPC4_ASRC_H__
-#define __SOF_IPC4_ASRC_H__
+#ifndef __SOF_ASRC_H__
+#define __SOF_ASRC_H__
 
 #include <stdint.h>
+/* If next define is set to 1 the ASRC is configured automatically. Setting
+ * to zero temporarily is useful is for testing needs.
+ */
+#define ASRC_AUTOARCH    1
+
+/* Select optimized code variant when xt-xcc compiler is used on HiFi3 */
+#if ASRC_AUTOARCH == 1
+#if defined __XCC__
+/* For xt-xcc */
+#include <xtensa/config/core-isa.h>
+#if XCHAL_HAVE_HIFI3 == 1
+/* Version for HiFi3 */
+#define ASRC_HIFI3	1
+#define ASRC_GENERIC	0
+#else
+/* Version for e.g. HiFi2EP */
+#define ASRC_HIFI3	0
+#define ASRC_GENERIC	1
+#endif
+#else
+/* For GCC */
+#define ASRC_GENERIC	1
+#define ASRC_HIFI3	0
+#endif /* XCC */
+#else
+/* Applied when ASRC_AUTOARCH is set to zero */
+#define ASRC_GENERIC	1 /* Enable generic */
+#define ASRC_HIFI3	0 /* Disable HiFi3  */
+#endif /* Autoarch */
+
+#if CONFIG_IPC_MAJOR_4
+
```

Comment:
This part is inconsistent, you are splitting the code in IPC4 and IPC3-specific parts, and create a common header which has IPC4-specific stuff.

If you truly want to have IPC-agnostic ASRC, this needs to be updated.

Reply:
thanks, done.